docs/predictive-analytics/next-best-offer/index.md

```diff
@@ -0,0 +1,208 @@
+```
+
+#### Creating the offer table
+
+The last lookup table you'll add is the `OFFERS` table:
+
+```sql
+CREATE TABLE offers (
+    OFFER_ID INTEGER PRIMARY KEY,
+    OFFER_NAME STRING,
+    OFFER_URL STRING
+) WITH (
+    KAFKA_TOPIC = 'OFFERS_STREAM',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+);
+```
+
+This table provides the enrichment information that will be needed when the application calculates the next best offer.
+
+To populate the `OFFERS` table, execute the following `INSERT` statements:
+
+```sql
+INSERT INTO offers (offer_id, offer_name, offer_url) VALUES (1,'new_savings','http://google.com.br/magnis/dis/parturient.json');
+INSERT INTO offers (offer_id, offer_name, offer_url) VALUES (2,'new_checking','https://earthlink.net/in/ante.js');
+INSERT INTO offers (offer_id, offer_name, offer_url) VALUES (3,'new_home_loan','https://webs.com/in/ante.jpg');
+INSERT INTO offers (offer_id, offer_name, offer_url) VALUES (4,'new_auto_loan','http://squidoo.com/venenatis/non/sodales/sed/tincidunt/eu.js');
+INSERT INTO offers (offer_id, offer_name, offer_url) VALUES (5,'no_offer','https://ezinearticles.com/ipsum/primis/in/faucibus/orci/luctus.html');
+```
+
+### Determining the next best offer
+
+Now you'll create the event stream that calculates the next best offer for your customers based on their activity. 
+
+#### Calculating the offer
+
+To perform the next offer calculation, create a stream that performs a join between the `CUSTOMER_ACTIVITY_STREAM` and the `CUSTOMERS` table:
+
+```sql
+CREATE STREAM next_best_offer
+WITH (
+    KAFKA_TOPIC = 'NEXT_BEST_OFFER',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+) AS
+SELECT 
+cask.ACTIVITY_ID,
+cask.CUSTOMER_ID as CUSTOMER_ID,
+cask.PROPENSITY_TO_BUY,
+cask.ACTIVITY_TYPE,
+ct.INCOME,
+ct.FICO,
+CASE  
+    WHEN ct.INCOME > 100000 AND ct.FICO < 700 AND cask.PROPENSITY_TO_BUY < 0.9 THEN 1
+    WHEN ct.INCOME < 50000 AND cask.PROPENSITY_TO_BUY < 0.9 THEN 2
+    WHEN ct.INCOME >= 50000 AND ct.FICO >= 600 AND cask.PROPENSITY_TO_BUY < 0.9 THEN 3
+    WHEN ct.INCOME > 100000 AND ct.FICO >= 700 AND cask.PROPENSITY_TO_BUY < 0.9 THEN 4
+    ELSE 5
+END AS OFFER_ID 
+FROM customer_activity_stream cask
+INNER JOIN customers ct ON cask.CUSTOMER_ID = ct.CUSTOMER_ID;
+```
+
+The `CASE` statement is the workhorse for the query. It provides the next offer for the customer based on information resulting from the join. Note that you're using an `INNER JOIN` here because if the customer id isn't found in the `CUSTOMERS` table, there's no calculation to make. You'll notice that the result of the `CASE` statement is a single integer—the code for the offer to be made—so there's one final step left.
+
+#### Final results
+
+For the last step, create a query which contains the final results by joining the `NEXT_BEST_OFFER` stream with the `OFFERS` table:
+
+```sql
+CREATE STREAM next_best_offer_lookup
```

Comment:
In this query we join on offer_id which implies a repartitioning of the input stream. We could avoid this cost, by adding a `PARTITION BY OFFER_ID` (well we would need to full case statement...) in the first query.

Not sure if it would confuse users... -- as an alternative, we could also merge both queries into one; frankly, it might not be less confusing as we would need to use the CASE statement in the second join condition, too...